pkg/virt-controller/watch/migration.go

```diff
@@ -283,6 +284,19 @@ func (md *MigrationController) execute(key string) error {
 			eventMsg := fmt.Sprintf("Finished migrating VM from %s to %s", vm.Status.NodeName, targetPod.Spec.NodeName)
 			vm.Status.NodeName = targetPod.Spec.NodeName
 			vm.Status.MigrationNodeName = ""
+			// Update SPICE host IP address after migration
+			node, err := md.clientset.CoreV1().Nodes().Get(targetPod.Spec.NodeName, metav1.GetOptions{})
+			if err != nil {
+				logging.DefaultLogger().Error().Reason(err).Msgf("fetching destination node %s failed", targetPod.Spec.NodeName)
+				return err
+			}
+			spiceHostAddr := networking.GetNodeInternalIP(node)
+			if spiceHostAddr == "" {
+				err := fmt.Errorf("fetching destination node %s internal IP failed", targetPod.Spec.NodeName)
+				logging.DefaultLogger().Error().Msg("VM node is unreachable")
+				return err
+			}
+			vm.Status.Graphics[0].Host = spiceHostAddr
```

Comment:
This (probably correctly) assumes there's only one Spice endpoint. I have a sneaking suspicion there's a corner case lurking here.

Reply:
hm I believe I will need to change it to the same way as it defined under `spiceFromVM` function, otherwise, we can have a problem
```
// TODO allow specifying the spice device. For now select the first one.
	for _, d := range vm.Status.Graphics {
		if d.Type == "spice" {
			spice := v1.NewSpice(vm.GetObjectMeta().GetNamespace(), vm.GetObjectMeta().GetName())
			spice.Info = v1.SpiceInfo{
				Type: "spice",
				Host: d.Host,
				Port: d.Port,
			}
			if spiceProxy != "" {
				spice.Info.Proxy = fmt.Sprintf("http://%s", spiceProxy)
			}
			return spice, nil
		}
```

Reply:
More important, the port can also be different after a migration. I think that the target virt-handler needs to update that information, after the migation succeeded. It is the only one with access to libvirt, to check on the domain xml what the new port is.

Reply:
alternatively - you can also think about defining the destination port upfront

Reply:
I believe set it automatically always better than leave it to a user, but I will need to check how can I do this.
@rmohr Can you help me a little with a flow, after which function under virt-controller, virt-handler takes control over VM update?

Reply:
I am not sure that would help us. I think it depends on the migration mode (post-copy vs pre-copy) when we can update the port and the IP, and I would want to avoid storing it on the cluster VM state for "later" use.

Reply:
@cynepco3hahue I think that the right point in time to update this is when the migration controller hands over the VM from one virt-handler to the other virt-handler and when virt-handler and when we have a domain XML of the Domain in our Domain cache on the target virt-handler. When we update the VM status, we can compare the reported Domain XML with the cluster wide VirtualMachine and update the port and the IP.

In general, I think that this not necessarely migration related. Whenever we detect a mismatch in the graphics sections, we should update it.